galite-core/src/main/kotlin/org/kopi/galite/ui/vaadin/visual/VFileHandler.kt

```diff
@@ -19,29 +19,276 @@
 package org.kopi.galite.ui.vaadin.visual
 
 import java.io.File
+import java.io.FileOutputStream
+import java.io.IOException
 
+import org.kopi.galite.ui.vaadin.upload.FileUploader
+import org.kopi.galite.visual.ApplicationContext
 import org.kopi.galite.visual.FileHandler
 import org.kopi.galite.visual.UWindow
 
 class VFileHandler : FileHandler() {
-  override fun chooseFile(window: UWindow, defaultName: String): File {
-    TODO("Not yet implemented")
+
+  // --------------------------------------------------
+  // FILE HANDLER IMPLEMENTATION
+  // --------------------------------------------------
+  override fun chooseFile(window: UWindow, defaultName: String): File? {
+    return chooseFile(window, null, defaultName)
+  }
+
+  override fun chooseFile(window: UWindow, dir: File?, defaultName: String): File? {
+    return try {
+      createTempFile(dir, defaultName)
+    } catch (e: IOException) {
+      e.printStackTrace()
+      getApplication().displayError(window, e.message)
+      null
+    }
   }
 
-  override fun chooseFile(window: UWindow, dir: File, defaultName: String): File? {
-    TODO("Not yet implemented")
+  override fun openFile(window: UWindow, defaultName: String): File? {
+    return openFile(window, null, defaultName)
```

Comment:
Better to use inline fcts instead
override fun openFile(window: UWindow, defaultName: String): File? = openFile(window, null, defaultName)